### PR TITLE
SONAR-9432 let rule search use OR instead of AND when filtering by tags

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java [hunk 1]
```diff
@@ -223,6 +223,26 @@ public void should_filter_on_organization_specific_tags() throws IOException {
   }
 
   @Test
+  public void when_searching_for_several_tags_combine_them_with_OR() throws IOException {
+    OrganizationDto organization = dbTester.organizations().insert();
+    RuleDefinitionDto both_tags_rule = createJavaRule();
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java]: Please use camel case for variables

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java [hunk 2]
```diff
@@ -280,6 +300,28 @@ public void should_return_organization_specific_tags() throws IOException {
   }
 
   @Test
+  public void should_not_return_tags_of_foreign_organization() throws IOException {
+    OrganizationDto organization_with_specific_tags = dbTester.organizations().insert();
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java]: Please use camel case for variables

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java [hunk 3]
```diff
@@ -280,6 +300,28 @@ public void should_return_organization_specific_tags() throws IOException {
...
+    OrganizationDto myOrganization = dbTester.organizations().insert();
+    RuleDefinitionDto rule = dbTester.rules().insert(setSystemTags("system1", "system2"));
+    insertMetadata(organization_with_specific_tags, rule, setTags("tag1", "tag2"));
+    indexRules();
+
+    SearchResponse result = ws.newRequest()
+      .setParam("facets", "tags")
+      .setParam("f", "tags")
+      .setParam("organization", myOrganization.getKey())
+      .executeProtobuf(SearchResponse.class);
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java]: Please add a space after this line to highlight what is tested

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java (hunk 1 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java]: Done.

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java (hunk 2 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java]: Done.

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java (hunk 1 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java]: Done.

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java (hunk 2 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java]: Done.

File: server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java (hunk 3 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java]: Done.

**Pull request merged**